NamedPipeWrapper/NamedPipeServer.cs

```diff
@@ -236,13 +276,39 @@ public void Stop()
 				}
 			}
 
+			if (_server != null && _server.IsConnected)
+			{
+				_server.Close();
+			}
+
 			// If background thread is still listening for a client to connect,
 			// initiate a dummy connection that will allow the thread to exit.
 			var dummyClient = new NamedPipeClient<TRead, TWrite>(_pipeName);
 			dummyClient.Start();
 			dummyClient.WaitForConnection(TimeSpan.FromSeconds(2));
 			dummyClient.Stop();
 			dummyClient.WaitForDisconnection(TimeSpan.FromSeconds(2));
+
+		}
+
+		public void Read()
+		{
+			try
+			{
+				if (_server.IsConnected && !_reader.EndOfStream)
+				{
+					string message = _reader.ReadLine();
+					if (message != null)
+					{
+						// Temporary - do something else with the command when we receive it
+						Console.WriteLine(String.Join("", message));
```

Comment:
How would this be used by the hosting service (e.g., Windows Service)? Should it emit an event?